api/src/context.rs

```diff
@@ -915,11 +947,64 @@ impl Context {
         version: u64,
         ledger_version: u64,
     ) -> Result<TransactionOnChainData> {
-        self.convert_into_transaction_on_chain_data(self.db.get_transaction_by_version(
-            version,
-            ledger_version,
-            true,
-        )?)
+        let mut txn = self.convert_into_transaction_on_chain_data(
+            self.db
+                .get_transaction_by_version(version, ledger_version, true)?,
+        )?;
+        if self
+            .node_config
+            .indexer_db_config
+            .enable_event_v2_translation
+        {
+            let _ = self.translate_v2_to_v1_events_for_version(version, &mut txn.events);
```

Comment:
why do we ignore result?

Reply:
The table info grpc service creates and uses their own instance of `Context` without `indexer_reader`, thus causing an expected error. I revised the code by adding additional condition such as `&& indexer_reader.is_some()`.